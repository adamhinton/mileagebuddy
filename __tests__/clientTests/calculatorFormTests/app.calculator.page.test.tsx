import CalculatorPage from "@/app/calculator/page";
import TestReduxStore from "@/app/utils/unitTestUtils/dummyReduxStore";
import { render } from "@testing-library/react";

describe("Sanity check", () => {
	it("should pass", () => {
		expect(1 + 1).toBe(2);
	});
});

describe("Calculator page", () => {
	it("Renders without errors", () => {
		render(
			<TestReduxStore>
				<CalculatorPage />
			</TestReduxStore>
		);
	});
});
