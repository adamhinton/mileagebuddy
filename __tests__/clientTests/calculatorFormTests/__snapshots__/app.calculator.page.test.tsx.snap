// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page matches snapshot from 4.30.2025 1`] = `
<div>
  <section
    class="min-h-screen p-4 sm:p-6 md:p-8 bg-neutral-50 dark:bg-neutral-900"
  >
    <div
      class="max-w-5xl mx-auto"
    >
      <header
        class="mb-6"
      >
        <h1
          class="text-2xl sm:text-3xl md:text-4xl font-bold text-neutral-800 dark:text-neutral-100"
        >
          Vehicle Information
        </h1>
        <p
          class="mt-2 text-neutral-600 dark:text-neutral-400"
        >
          Input your vehicle information to calculate your true cost per mile.
        </p>
      </header>
      <form
        class="mileage-calc-container"
      >
        <div
          class="mileage-calc-form-header"
        >
          <h1
            class="mileage-calc-form-title"
          >
            Create New Vehicle
          </h1>
          <button
            aria-label="Clear Form"
            class="px-4 py-2 rounded-md transition-colors focus:outline-none focus:ring-2 
          text-sm sm:text-base bg-background-elevated text-neutral-text border border-primary-100 hover:bg-background-highlight focus:ring-primary-200 dark:border-primary-200 self-start"
            type="button"
          >
            Clear Form
          </button>
        </div>
        <div
          class="mileage-calc-form-radio-group-container"
        >
          <h3
            class="mileage-calc-form-radio-group-title"
          >
            Vehicle Type
          </h3>
          <div
            class="mileage-calc-form-radio-group"
          >
            <label
              class="mileage-calc-form-radio-label "
            >
              <input
                class="mileage-calc-form-radio-input"
                name="type"
                type="radio"
                value="gas"
              />
              <span
                class="mileage-calc-form-radio-text"
              >
                Gas / Standard Hybrid
              </span>
            </label>
            <label
              class="mileage-calc-form-radio-label "
            >
              <input
                class="mileage-calc-form-radio-input"
                name="type"
                type="radio"
                value="electric"
              />
              <span
                class="mileage-calc-form-radio-text"
              >
                Electric
              </span>
            </label>
            <label
              class="mileage-calc-form-radio-label "
            >
              <input
                class="mileage-calc-form-radio-input"
                name="type"
                type="radio"
                value="hybrid"
              />
              <span
                class="mileage-calc-form-radio-text"
              >
                Plug-in Hybrid
              </span>
            </label>
          </div>
        </div>
        <div
          class="mileage-calc-form-footer"
        >
          <button
            aria-label="Save Vehicle"
            class="px-4 py-2 rounded-md transition-colors focus:outline-none focus:ring-2 
          text-sm sm:text-base bg-primary text-white hover:bg-primary-600 focus:ring-primary-200 px-6 py-2.5"
            type="submit"
          >
            Save Vehicle
          </button>
        </div>
      </form>
    </div>
  </section>
</div>
`;
